### ISSUE TITLE
every(mapBy()) does not recalculate when mapped properties change

### ISSUE THREAD

**Issue by lolmaus** [kellyselden/ember-awesome-macros#263]
So I have a CP like this:

```js
  areAllShorterThanFive_cp: Ember.computed('data.@each.name', function () {
    return this
    	.get('data')
    	.mapBy('name')
    	.every(name => name.length < 5)
  }),  
```

I've assumed it can be rewritten with `ember-awesome-macros` like this:

```js
  areAllShorterThanFive_every_mapBy:
    every(
      mapBy('data', 'name'),
      item => item.get('name').length < 5
    ),
```

But the latter does not recalculate. :(

Here's a demo: https://ember-twiddle.com/0dc43e01f156470d6c427f856963e0f4?openFiles=controllers.application.js%2C

I also tried this:

```js
  areAllShorterThanFive_every: every('data.@each.name',
    item => item.get('name').length < 5
  ),
```

but it produces a warning "You used the key "data.@each.name.[]" which is invalid." and still does not work.

We definitely need an `everyBy` macro, but why doesn't `every(mapBy())` work? Is there a way to make it work? Either it's a bug or I'm missing something important about the addon. I wonder how many issues like this I already have in my apps.

**Comment by mriska** [comment]
I think this is the same issue as reported in #164 ?

**Comment by lolmaus** [comment]
It is, thank you.

**Comment by kellyselden** [comment]
We have some array issues I'm probably going to fix this weekend. But your macro usage also has some issues. It would be written:

```js
  areAllShorterThanFive_every_mapBy:
    every(
      mapBy('data', raw('name')),
      name => name.length < 5
    ),
```

**Comment by lolmaus** [comment]
@kellyselden Thank you for pointing at the bug.

I've fixed it like you said and it still won't work: 

https://ember-twiddle.com/0dc43e01f156470d6c427f856963e0f4?openFiles=controllers.application.js%2C

**Comment by kellyselden** [comment]
@lolmaus You're right. You are affected by the bug. Sorry. Still looking to fix this weekend. The fix is known #252.
